lib/python/picongpu/plugins/plot_mpl/transition_radiation_visualizer.py

```diff
@@ -0,0 +1,275 @@
+"""
+This file is part of PIConGPU and based on energy_histogram_visualizer.py.
+
+Authors: Finn-Ole Carstens, Sebastian Starke
+"""
+
+from picongpu.plugins.data import TransitionRadiationData
+from picongpu.plugins.plot_mpl.base_visualizer import Visualizer as \
+    BaseVisualizer
+from matplotlib.ticker import FixedLocator
+import numpy as np
+import scipy.constants as const
+
+
+class Visualizer(BaseVisualizer):
+    """
+    Class for creation of different plots for analysis of the CTR plugin in
+    PIConGPU.
+    """
+
+    def __init__(self, run_directories=None, ax=None):
+        """
+        Parameters
+        ----------
+        run_directories: list of tuples of length 2
+            or single tuple of length 2.
+            Each tuple is of the following form (sim_name, sim_path)
+            and consists of strings.
+            sim_name is a short string used e.g. in plot legends.
+            sim_path leads to the run directory of PIConGPU
+            (the path before ``simOutput/``).
+            If None, the user is responsible for providing run_directories
+            later on via set_run_directories() before calling visualize().
+        ax: matplotlib.axes
+        """
+        super().__init__(TransitionRadiationData, run_directories, ax)
+        self.type = None
+
+    def _create_plt_obj(self, idx):
+        """
+           Implementation of base class function.
+           Turns 'self.plt_obj' into a matplotlib.pyplot.plot object.
+           """
+        if self.type == "spectrum":
+            omegas, spectral_power = self.data[idx]
+            self.plt_obj[idx] = self.ax.plot(omegas, spectral_power)
+        elif self.type == "sliceovertheta":
+            thetas, spectral_power = self.data[idx]
+            self.plt_obj[idx] = self.ax.plot(thetas, spectral_power)
+        elif self.type == "sliceoverphi":
+            phis, spectral_power = self.data[idx]
+            self.plt_obj[idx] = self.ax.plot(phis, spectral_power)
+        elif self.type == "heatmap":
+            theta_mesh, phi_mesh, spectral_power = self.data[idx]
+            im = self.plt_obj[idx] = self.ax.pcolormesh(theta_mesh, phi_mesh,
+                                                        spectral_power)
+            self.ax.get_figure().colorbar(im,
+                                          label=r"Spectral Power $d^2 W / "
+                                                r"d\omega d\Omega$")
+
+    def _update_plt_obj(self, idx):
+        """
+        Implementation of base class function.
+        """
+        # counts, bins, iteration, dt = self.data[idx]
+        # self.plt_obj[idx].set_data(bins, counts)
+        # TODO
```

Comment:
Here seems to be something incomplete. 
@finnolec Could you please comment on what should be added here and whether this will be done with this pull request or a follow up.